### PR TITLE
Disallow periodic sync registrations from the background.

### PR DESCRIPTION
**Pull request by mugdhalakhani**
This also updates some links to address the linking errors bikeshed emits
after being updated.

@beverloo @jakearchibald PTAL.

### CODE REVIEW

File: spec/PeriodicBackgroundSync-index.bs [hunk 1]
```diff
@@ -215,6 +217,10 @@ The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">reg
 1. If |serviceWorkerRegistration|’s [=active worker=] is null, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
 1. Else, [=enqueue the following steps=] to the [=periodicsync processing queue=]:
     1. If the {{PermissionState}} for a {{PermissionDescriptor}} with {{PermissionDescriptor/name}} <code>"periodic-background-sync"</code> is not {{PermissionState/granted}}, [=reject=] |promise| with a {{NotAllowedError}} and abort these steps.
+    1. Let |isBackground|, a boolean, be true.
+    1. For each |client| in the [=service worker clients=] for the |serviceWorkerRegistration|'s [=origin=]:
+      1. If |client|'s [=frame type=] is top-level of auxillary, set |isBackground| to false.
```
Comment (beverloo) [inline spec/PeriodicBackgroundSync-index.bs]: of -> or

Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs]: oops, fixed!

**Comment by mugdhalakhani** [comment]
> It's unfortunate that link-defauls keeps growing, even for something in WebIDL. Do we really need that?

The WebIDL reference has replaced a reference to promises-guide. So there's no net increase in link defaults because of that. I've tried running bikeshed using curl to rule out any staleness in my local bikeshed references, which means that these additions are indeed necessary to disambiguate between two references of the same name.

Without the 'resolve' default-link, I see this error:
Arbitrarily chose https://w3c.github.io/FileAPI/#blob-url-resolve
To auto-select one of the following refs, insert one of these lines into a <pre class=link-defaults> block:
spec:webidl; type:dfn; text:resolve
spec:fileapi-1; type:dfn; text:resolve

and without the "service worker" default-link, I see this error:
Arbitrarily chose https://w3c.github.io/ServiceWorker/#dfn-service-worker
To auto-select one of the following refs, insert one of these lines into a <pre class=link-defaults> block:
spec:service-workers; type:dfn; for:ServiceWorkerGlobalScope; text:service worker
spec:service-workers; type:dfn; for:/; text:service worker

Why do you think link-defaults are bad? I understand the concerns around adding to bikeshed anchors, but I'd like to understand your hesitation around adding to link-defaults. Happy to try another recommended way to disambiguate here. :)

**Comment by mugdhalakhani** [comment]
Thanks Peter. I'll ask around to see if there's a way around disambiguating with link defaults.

**Pull request merged**